### PR TITLE
Allow creating multiple workspaces from the single sample

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

This PR main goal is to allow users to create multiple workspaces from the same sample. To get this I added a new step to the Factory flow that handles workspace names conflict and prompts the user whether to switch to an existing workspace or to proceed to create a new workspace (appending a suffix to its name):

Screenshot 2022-09-06 at 14 38 00

#### Opening the existing workspace:

https://user-images.githubusercontent.com/16220722/189610353-55cd653a-85f8-44bf-a00d-9a662a2a28f7.mov

#### Creating a new workspace:

https://user-images.githubusercontent.com/16220722/189610650-ffc88685-98c7-4f16-9fed-3bf29f37e97d.mov

Additionally, this PR changes a bit the workspace loading flow: it adds a new step to check if the number of simultaneously running workspaces is within the limit:

https://user-images.githubusercontent.com/16220722/189612766-d885bb7b-bb5e-40fd-94cb-ec2fee16d78d.mov

### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/21561

### Is it tested? How?

1. Try to create multiple workspaces from a single sample (or devfile)
2. You should be prompted to switch to an existing workspace or to continue creating a new one

You can test this PR here: https://eclipse-che.apps.cluster-8rnq4.8rnq4.sandbox506.opentlc.com/, or use your Eclipse Che installation with this Dashboard image **quay.io/eclipse/che-dashboard:pr-622**.

### CODE REVIEW

File: packages/dashboard-frontend/src/containers/Loader/Factory/Steps/Apply/Resources/prepareResources.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    devWorkspace.spec.template.attributes[DEVWORKSPACE_STORAGE_TYPE] = storageType;
+  } else if (devWorkspace.spec?.template?.attributes?.[DEVWORKSPACE_STORAGE_TYPE]) {
+    delete devWorkspace.spec.template.attributes[DEVWORKSPACE_STORAGE_TYPE];
+  }
+
+  return [devWorkspace, devWorkspaceTemplate];
+}
+
+function addSuffix(devWorkspace: DevWorkspace, devWorkspaceTemplate: DevWorkspaceTemplate) {
+  const suffix = generateWorkspaceName('');
```
Comment (olexii4) [inline packages/dashboard-frontend/src/containers/Loader/Factory/Steps/Apply/Resources/prepareResources.ts:59]: Could you create a separate method for generating sufixes?

**Review by olexii4 — APPROVED** [review]
I tested it with https://eclipse-che.apps.cluster-8rnq4.8rnq4.sandbox506.opentlc.com. All works as expected.
Good job.

**Review by ibuziuk — APPROVED** [review]
LGTM, +1 to promote to the dogfooding

File: packages/dashboard-frontend/src/containers/Loader/Factory/Steps/Apply/Resources/prepareResources.ts (hunk 1 above)
Comment (akurinnoy) [reply packages/dashboard-frontend/src/containers/Loader/Factory/Steps/Apply/Resources/prepareResources.ts]: done!

**Pull request merged**